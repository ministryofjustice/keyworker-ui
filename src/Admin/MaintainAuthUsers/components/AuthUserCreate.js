import React from 'react'
import PropTypes from 'prop-types'
import InputField from '@govuk-react/input-field'
import Button from '@govuk-react/button'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import { lookupMeta } from '../../../govuk-helpers'
import { errorType } from '../../../types'

const AuthUserCreate = ({ handleChange, handleCreate, error }) => (
  <GridRow>
    <GridCol setWidth="two-thirds">
      <form onSubmit={handleCreate}>
        <InputField
          meta={lookupMeta('username', error)}
          htmlFor="username"
          mb={6}
          input={{
            id: 'username',
            name: 'username',
            onChange: handleChange,
            autoComplete: 'create-username',
            spellcheck: false,
          }}
        >
          Username
        </InputField>
        <InputField
          meta={lookupMeta('email', error)}
          htmlFor="email"
          mb={6}
          input={{
            id: 'email',
            name: 'email',
            onChange: handleChange,
            autoComplete: 'create-email',
            type: 'email',
            spellcheck: false,
          }}
        >
          Email address
        </InputField>
        <InputField
          meta={lookupMeta('firstName', error)}
          htmlFor="firstName"
          mb={6}
          input={{ id: 'firstName', name: 'firstName', onChange: handleChange, spellcheck: false }}
        >
          First name
        </InputField>
        <InputField
          meta={lookupMeta('lastName', error)}
          htmlFor="lastName"
          mb={6}
          input={{ id: 'lastName', name: 'lastName', onChange: handleChange, spellcheck: false }}
        >
          Last name
        </InputField>
        <Button type="submit" id="create-button">
          Create
        </Button>
      </form>
    </GridCol>
  </GridRow>
)

AuthUserCreate.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleCreate: PropTypes.func.isRequired,
  error: errorType,
}

AuthUserCreate.defaultProps = {
  error: [],
}

export default AuthUserCreate
