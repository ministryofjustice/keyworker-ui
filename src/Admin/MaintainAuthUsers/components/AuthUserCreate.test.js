import React from 'react'
import renderer from 'react-test-renderer'
import AuthUserCreate from './AuthUserCreate'

const groups = [
  { groupCode: 'GROUP_1', groupName: 'Group 1' },
  { groupCode: 'GROUP_2', groupName: 'Group 2' },
  { groupCode: 'GROUP_3', groupName: 'Group 3' },
]

describe('Auth Create form', () => {
  const stubFunc = () => {}
  describe('Form', () => {
    it('should render correctly', () => {
      const wrapper = renderer
        .create(<AuthUserCreate handleChange={stubFunc} handleCreate={stubFunc} groupList={[]} />)
        .toJSON()
      expect(wrapper).toMatchSnapshot()
    })

    it('should render correctly with groups', () => {
      const wrapper = renderer
        .create(<AuthUserCreate handleChange={stubFunc} handleCreate={stubFunc} groupList={groups} />)
        .toJSON()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
