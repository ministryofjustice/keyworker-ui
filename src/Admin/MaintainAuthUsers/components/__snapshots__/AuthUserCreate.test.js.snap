// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Create form Form should render correctly 1`] = `
<div
  className="sc-gzVnrw lauuhu"
>
  <div
    className="sc-htoDjs iFTIaU"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="sc-bdVaJa cmpKXw"
        htmlFor="username"
      >
        <span
          className="sc-bwzfXH lbxcXc"
        >
          Username
        </span>
        <input
          autoComplete="create-username"
          className="sc-ifAKCX hTyiys"
          id="username"
          name="username"
          onChange={[Function]}
          type="text"
        />
      </label>
      <label
        className="sc-bdVaJa cmpKXw"
        htmlFor="email"
      >
        <span
          className="sc-bwzfXH lbxcXc"
        >
          Email address
        </span>
        <input
          autoComplete="create-email"
          className="sc-ifAKCX hTyiys"
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
        />
      </label>
      <label
        className="sc-bdVaJa cmpKXw"
        htmlFor="firstName"
      >
        <span
          className="sc-bwzfXH lbxcXc"
        >
          First name
        </span>
        <input
          className="sc-ifAKCX hTyiys"
          id="firstName"
          name="firstName"
          onChange={[Function]}
          type="text"
        />
      </label>
      <label
        className="sc-bdVaJa cmpKXw"
        htmlFor="lastName"
      >
        <span
          className="sc-bwzfXH lbxcXc"
        >
          Last name
        </span>
        <input
          className="sc-ifAKCX hTyiys"
          id="lastName"
          name="lastName"
          onChange={[Function]}
          type="text"
        />
      </label>
      <button
        className="sc-EHOje hEaLol"
        disabled={false}
        id="create-button"
        type="submit"
      >
        Create
      </button>
    </form>
  </div>
</div>
`;
